### ISSUE TITLE
Set up Asciidoctor Core docs component

### ISSUE THREAD

**Issue by graphitefriction** [asciidoctor/asciidoctor#2651]
Set up a documentation component in preparation for semantic versioning, dependency deprecation, and the 2.0 roadmap. I'm hitting the big red button, @mojavelinux, so buckle up :smiling_imp: !

Done criteria:

- add antora.yml, version is 'dev' since these docs don't correspond to a current release
- add basic nav.adoc so when site is built locally the pages are easy to test
- add ROOT module
- add setup module (this would contain the system requirements, install instructions, upgrade instructions, and relate materials - the name of the module is under consideration)
- add index page with note that these aren't the real docs, there over at asciidoctor.org still
- add platforms.adoc
  - this page describes the supported operating systems and Ruby versions (proposed for 1.5.7 - 2.0)
- add versions-and-lifecycle-policies.adoc
  - the semantic versioning is coming and we'll abide like the dude :bowling:  
- add upgrade-syntax.adoc 
  - some syntax is retiring, some is mutating (proposed for 1.5.7 - 2.0)

Open questions:

- Do we want to merge this into master or keep it in a separate branch? (currently staged for master #2652)
- I've built this site on my computer using a local playbook, if you want me to post/commit the playbook somewhere specific, let me know.

**Comment by graphitefriction** [comment]
I'm experimenting with structures right now and leaning heavily on the side of over-moduleing (and making up words) since it's easy to recombine and move modules as we settle on a directory structure. To that purpose, I've added a couple of "pages" and "partials" from the user manual (lists, source blocks, syntax highlighting). I've also smacked my staging playbook `site.yml` into the root of the repo temporarily, so if anyone else wants to generate the current experiment they can.
(I've moved a lot of AsciiDoc attributes and shared external URLs into the playbook, which is why you'll need it to do the generation).

I'm going to continue to push in some atypical pages over the next few days as I continue to rearrange the source material (keeping in mind that where the source material is located has no bearing on the published sidebar navigation/component drawer list). I'm looking for the best groupings of pages and partials in order to keep the page IDs and includes clean-ish.
